src/actions/arcActions.ts

```diff
@@ -674,19 +665,37 @@ export function createProposal(daoAvatarAddress: string, title: string, descript
       // Go back to home page while action create proposal operation gets carried out
       dispatch(push("/dao/" + daoAvatarAddress));
 
-      await contributionRewardInstance.proposeContributionReward({
-        title,
-        avatar: daoAvatarAddress,
-        beneficiaryAddress,
-        description,
-        ethReward: Util.toWei(ethReward),
-        externalToken: dao.externalTokenAddress ? dao.externalTokenAddress : "",
-        externalTokenReward: dao.externalTokenAddress && externalTokenReward ? Util.toWei(externalTokenReward) : 0,
-        nativeTokenReward: Util.toWei(nativeTokenReward),
-        numberOfPeriods: 1,
-        periodLength: 0,
-        reputationChange: Util.toWei(reputationReward),
-      } as any);
+      const dao = getState().arc.daos[daoAvatarAddress];
+      const contributionRewardInstance = await Arc.ContributionRewardFactory.deployed();
+      const wallet = await getWallet()
+      const contributionReward = new ethers.Contract(contributionRewardInstance.address, contributionRewardArtifacts.abi, wallet)
```

Comment:
adding ethers just for this seems unneccesary, you can get the truffle contract through contributionRewardInstance.contract and use that

Reply:
Agree with Tibet here. Or use web3 (< 1.0 comes with arc.js)

Reply:
Good suggestion, I like this idea better